test/test_core.py

```diff
@@ -9761,6 +9761,14 @@ def test_main_reads_args(self):
 
   @requires_node
   def test_promise(self):
+    # This test depends on Promise.any, which in turn requires a modern target.  Check that it
+    # fails to even build without bumping the min versions:
+    err = self.expect_fail([EMCC, test_file('core/test_promise.c')])
```

Comment:
Don't all the versions below support Promise.any, so this build should succeed rather than fail? I must be missing something.

Reply:
The default versions fail, so we run a compile here to check that it fails, then we set the correct settings and then we expect the test to pass, right?

Reply:
Aha, yes, makes sense. I didn't grok that this was doing the build twice, once before and once after setting the settings.